status/reporters.go

```diff
@@ -480,6 +480,21 @@ func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx co
 
 		for _, integrationTestStatusDetail := range integrationTestStatusDetails {
 			integrationTestStatusDetail := *integrationTestStatusDetail // G601
+			latestUpdateTime, err := gitops.GetLatestUpdateTime(snapshot)
+			if err != nil {
+				logger.Error(err, "failed to get lates update annotation for snapshot",
+					"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+				return err
+			}
+			timeDiff := latestUpdateTime.Sub(integrationTestStatusDetail.LastUpdateTime)
+			if timeDiff < 0 {
+				err := metadata.SetAnnotation(snapshot, gitops.SnapshotPRLastUpdate, integrationTestStatusDetail.LastUpdateTime.String())
+				if err != nil {
+					logger.Error(err, "failed to set annotation for snapshot",
+						"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+					return err
+				}
```

Comment:
What's the purpose if this? It has no effect to code execution, it just update timestamp and still executes reporting into GH

Reply:
Updated